client/my-sites/themes/theme-showcase.jsx

```diff
@@ -292,15 +305,9 @@ class ThemeShowcase extends Component {
 	};
 
 	onTierSelect = ( { value: tier } ) => {
-		// Due to the search backend limitation, static filters other than "All"
-		// can only have "All" tier.
-		const staticFilters = this.getStaticFilters();
-		if (
-			tier !== 'all' &&
-			this.isStaticFilter( this.state.tabFilter ) &&
-			this.state.tabFilter.key !== staticFilters.ALL.key
-		) {
-			this.setState( { tabFilter: staticFilters.ALL } );
+		// Due to the search backend limitation, "My Themes" can only have "All" tier.
+		if ( tier !== 'all' && this.state.tabFilter.key === staticFilters.MYTHEMES.key ) {
+			this.setState( { tabFilter: staticFilters.RECOMMENDED } );
```

Comment:
Is the limitation mentioned in the comment isn't true anymore? Maybe we should update or remove the comment then?

Reply:
The limitation is still true. You cannot use other tier than "All" in "My Themes", so we automatically activate the "Recommended" tab when a non-supported tier is selected.